internal/controller/replication.storage/volumereplication_controller.go

```diff
@@ -741,13 +741,19 @@ func (r *VolumeReplicationReconciler) enableReplication(vr *volumeReplicationIns
 
 	resp := volumeReplication.Enable()
 
-	if resp.Error != nil {
-		vr.logger.Error(resp.Error, "failed to enable volume replication")
+	if resp.Error == nil {
+		return nil
+	}
 
-		return resp.Error
+	vr.logger.Error(resp.Error, "failed to enable volume replication")
+
+	if resp.HasKnownGRPCError(enableReplicationKnownErrors) {
+		setFailedValidationCondition(&vr.instance.Status.Conditions, vr.instance.Generation)
+	} else {
+		setFailedPromotionCondition(&vr.instance.Status.Conditions, vr.instance.Generation)
```

Comment:
can you please use `setFailureCondition(instance)` as it internally takes care of setting the same.

Reply:
we already have a generic helper function to set the conditions i don't like to see others calling it even though we have a helper function to do it. and for this again it's a bug we should not be setting `setFailedPromotionCondition` condition if the enable replication fails, the condition should reflect what went wrong.

Reply:
> we already have a generic helper function to set the conditions i don't like to see others calling it even though we have a helper function to do it. and for this again it's a bug we should not be setting `setFailedPromotionCondition` condition if the enable replication fails, the condition should reflect what went wrong.

Enabling replication is part of promotion. I don't see a issue with that part.

Why do you want to force a helper function that is used at a lot of other places to fit into a new scenario ?
If we do that, we'll need more arguments and the function will not be so generic anymore.

Reply:
even thought its part of the condition we should reflect on what exactly went wrong https://github.com/csi-addons/kubernetes-csi-addons/issues/666 is the tracker for it.

>Why do you want to force a helper function that is used at a lot of other places to fit into a new scenario ?
If we do that, we'll need more arguments and the function will not be so generic anymore.

i don't see any new change in the behaviour to call `setFailureCondition` instead of `setFailedPromotionCondition`

Reply:
I really don't get why you want to call that function when you already know which function it will in turn call.

There are a lot of places already calling functions from status.go directly.